src/ui/public/notify/notifier.js

```diff
@@ -39,22 +34,61 @@ function closeNotif(cb, key) {
     // this === notif
     var i = notifs.indexOf(this);
     if (i !== -1) notifs.splice(i, 1);
-    if (this.timerId) this.timerId = clearTO(this.timerId);
+    if (this.timerId) {
+      Notifier.config.clearInterval(this.timerId);
+      this.timerId = undefined;
+    }
     if (typeof cb === 'function') cb(key);
   };
 }
 
+function cancelNotifTimer(cb, key) {
+  return function () {
+    if (this.timerId) {
+      Notifier.config.clearInterval(this.timerId);
+      this.timerId = undefined;
+    }
+
+    if (typeof cb === 'function') cb(key);
+  };
+}
+
+/**
+ * Initiates a timer to update _timeRemaining_ on the notif at second
+ * intervals and clears the notif once the notif _lifetime_ has been reached.
+ */
+function startNotifTimer(cb) {
+  var notif = this;
+  var interval = 1000;
+
+  if (notif.lifetime === Infinity) return;
+
+  notif.timeRemaining = Math.floor(notif.lifetime / interval);
+
+  notif.timerId = Notifier.config.setInterval(function () {
+    notif.timeRemaining -= 1;
+
+    if (notif.timeRemaining === 0) {
+      closeNotif(cb, 'ignore').call(notif);
+    }
+  }, interval, notif.timeRemaining);
+
+  notif.cancelTimer = cancelNotifTimer(cb);
+}
+
+function restartNotifTimer(cb) {
+  var notif = this;
```

Comment:
I like that a lot better and updated some of the other functions for consistency.